cmake/Platform/Mac/runtime_install_mac.cmake.in

```diff
@@ -7,8 +7,13 @@
 #
 
 cmake_path(SET file_path "${CMAKE_INSTALL_PREFIX}/@install_relative_binaries_path@/@runtime_output_filename@")
+cmake_path(GET file_path EXTENSION LAST_ONLY file_ext)
 
-if(@runtime_output_filename@ MATCHES ".app")
+if(file_ext STREQUAL .app)
+
+    file(INSTALL @CMAKE_BINARY_DIR@/runtime_install/$<CONFIG>/BinariesInstallPath.setreg
```

Comment:
Since this setreg only makes sense for the installed version of the bundle, I figured this is a better place to put it in the bundle(during the install process before codesigning) instead of making it a runtime dependency during the build process.

Reply:
Does both this template file and the the `PreInstall_mac.cmake.in` template need the same logic for copying over the BinariesInstall.setreg file

Now that the O3DE_SDK is a target, should the `PreInstallSteps_mac.cmake.in` be in the BundleLauncher CMakeLists.txt?

Reply:
The O3DE_SDK is an executable right now. The bundle is constructed during install. If we make it an app during build, then I guess we could reuse the runtime_install cmake generated for the app. I'm okay with doing that but I would prefer to do that in a separate PR once this is in. The reason is we would still need the `PreInstallSteps_mac.cmake.in` because O3DE_SDK.app would still need special handling. It needs to be copied there before everything else. This is why call install(SCRIPT) on `PreInstallSteps_mac.cmake.in` at the top of `Install_mac.cmake` which is included from our root CMakeLists.txt. If any target gets added before this(even if we don't call install(TARGET) on the targets yet), CMake will add a call to `include(${target_dir}/cmake_install.cmake)` before the call to include our PreInstallSteps. So some stuff ends up outside the SDK app bundle. We also download the python framework for the engine at the end. So the codesign for the O3DE_SDK.app bundle also needs to happen at the very end. Hence the post install steps.

Reply:
After more thought, I don't think it's possible to use the runtime_install cmake generated for the O3DE_SDK.app even if we build it as app. This is because the file_path that's being set in this generated file will be pointing to O3DE_SDK.app that's within the install(i.e. `O3DE_SDK.app/Contents/Engine/bin/${PAL_PLATFORM_NAME}/$<CONFIG>/${LY_BUILD_PERMUTATION}/O3DE_SDK.app`). We want to put the setreg file at `O3DE_SDK.app/Contents/MacOS/Registry`. If we want to use the generated runtime_install cmake, we'd have to special case it for the outer-most bundle. It seems cleaner to do that in the PreInstallSteps_mac.cmake.